eZ/Publish/Core/FieldType/Time/SearchField.php

```diff
@@ -28,10 +29,17 @@ class SearchField implements Indexable
      */
     public function getIndexData(Field $field, FieldDefinition $fieldDefinition)
     {
+        if ($field->value->data !== null && $field->value->data !== '') {
```

Comment:
`if ($field->value->data !== null)` is enough because `TimeConverter` also [ensures](https://github.com/ezsystems/ezpublish-kernel/blob/v6.7.3/eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/TimeConverter.php#L55-L63) `data` remains `null` if not set.